02_Util/src/networkconnection/authenticator/NetworkProfileFilter.ts

```diff
@@ -0,0 +1,91 @@
+        deviceModelRepository: IDeviceModelRepository,
+        logger?: Logger<ILogObj>,
+    ) {
+        super(logger);
+        this._deviceModelRepository = deviceModelRepository;
+    }
+
+    protected shouldFilter(options: AuthenticationOptions): boolean {
+        return true;
+    }
+
+    protected async filter(
+        identifier: string,
+        request: IncomingMessage,
+        options: AuthenticationOptions
+    ): Promise<void> {
+        const isConfigurationSlotAllowed = await this._isConfigurationSlotAllowed(identifier, options.securityProfile);
+        if (!isConfigurationSlotAllowed) {
+            throw Error(`SecurityProfile not allowed ${options.securityProfile}`);
+        }
+    }
+
+    private async _isConfigurationSlotAllowed(identifier: string, securityProfile: number) {
+        const r = await this._deviceModelRepository
+            .readAllByQuerystring({
+                stationId: identifier,
+                component_name: 'OCPPCommCtrlr',
+                variable_name: 'NetworkConfigurationPriority',
+                type: AttributeEnumType.Actual,
+            })
+        if (r && r[0]) {
+            const configurationSlotsString = r[0].value;
+            if (configurationSlotsString && configurationSlotsString.trim() !== '') {
+                // Split the string by commas to get an array of string numbers
+                const configurationSlotStringsArray = configurationSlotsString.split(',');
+                // Parse the array into numbers and filter out the rest
+                const configurationSlotsArray = configurationSlotStringsArray.map(configurationSlotString => parseInt(configurationSlotString, 10))
+                    .filter(configurationSlotId => {
+                        if (isNaN(configurationSlotId)) {
+                            this._logger.error("NetworkConfigurationPriority elements must be integers: " + configurationSlotsString);
+                            return false;
+                        } else {
+                            return true;
+                        }
+                    });
+
+                if (configurationSlotsArray.length == 0) {
+                    this._logger.debug("No valid configuration slots to check: " + configurationSlotsString);
+                    return true;
+                } else {
+                    let securityProfileAllowed = false;
+                    for (const configurationSlot of configurationSlotsArray) {
+                        const chargingStationNetworkProfile = await ChargingStationNetworkProfile.findOne({ where: { stationId: identifier, configurationSlot: configurationSlot } });
+                        if (chargingStationNetworkProfile) {
+                            const serverNetworkProfile = await ServerNetworkProfile.findByPk(chargingStationNetworkProfile.websocketServerConfigId);
+                            if (serverNetworkProfile && securityProfile >= serverNetworkProfile.securityProfile) {
```

Comment:
in the scenario where a security profile is found to be greater than a configuration slot (and possibly all existing configuration slots), is there a need to store that security profile as well as a configuration? or is that handled otherwise in the SetNetworkProfile handler in the configuration api?